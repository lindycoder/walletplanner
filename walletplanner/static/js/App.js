import React from 'react';
import AddTransactions from './AddTransactions';
import TransactionsView from './TransactionsView';
import {StaticMonthly} from './Periodicity';
import {moment} from './utils';
import PropTypes from 'prop-types';

class App extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            transactions: []
        }
    }

    render() {
        return (
            <div className="App container">
                <h1 className="display-4">Wallet Planner</h1>
                <AddTransactions server={this.props.server}
                                 onTransactionAdded={this.refreshTransactions.bind(this)}/>
                <TransactionsView transactions={this.state.transactions}
                                  periodicity={new StaticMonthly([1, 15])}
                                  currentDate={moment().startOf('day')}
                                  periodStart={-3}
                                  openingBalance={0}/>
            </div>
        )
    }

    refreshTransactions() {
        this.props.server.getTransactions()
            .then(ts => this.setState({transactions: ts}));
    }
}

App.propTypes = {
    homeUrl: PropTypes.string.isRequired,
    echoUrl: PropTypes.string.isRequired,
    server: PropTypes.object.isRequired,
};

export default App;
