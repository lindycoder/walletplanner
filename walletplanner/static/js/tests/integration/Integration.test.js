import React from 'react';
import { mount } from 'enzyme';
import App from '../../App';
import FakeServer from './FakeServer';

describe('App', () => {
    let props;
    let wrapper;

    beforeEach(() => {
        props = {
          homeUrl: '/home-url',
          echoUrl: '/echo-url',
          server: new FakeServer(),
        };
        wrapper = mount(<App {...props} />);
    });

    it('should be able to add and see a transaction', async () => {
        let transactionAmountInput = wrapper.find('input').first();
        transactionAmountInput.instance().value = '1.00';
        transactionAmountInput.simulate('change');

        wrapper.find('button').simulate('click');

        await undefined;
        wrapper.instance().forceUpdate();
        await undefined;
        wrapper.update();

        expect(wrapper.find('.transaction-badge-amount').text()).toBe("1.00$")
    });
});
