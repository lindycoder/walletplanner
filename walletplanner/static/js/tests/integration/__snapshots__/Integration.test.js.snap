// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should be able to add and see a transaction 1`] = `
<App
  echoUrl="/echo-url"
  homeUrl="/home-url"
  server={
    FakeServer {
      "transactions": Array [
        Object {
          "amount": 100,
          "category": "",
          "description": "",
        },
      ],
    }
  }
>
  <div
    className="App container"
  >
    <h1
      className="display-4"
    >
      Wall Planner
    </h1>
    <AddTransactions
      onTransactionAdded={[Function]}
      server={
        FakeServer {
          "transactions": Array [
            Object {
              "amount": 100,
              "category": "",
              "description": "",
            },
          ],
        }
      }
    >
      <div>
        Add transactions
        <TransactionLine
          transaction={
            Object {
              "amount": 100,
              "category": "",
              "description": "",
            }
          }
        >
          <div
            className="form-row"
          >
            <div
              className="form-group col-sm-11"
            >
              <DatePicker
                allowSameDay={false}
                className="transaction-line-date"
                dateFormat="MM/dd/yyyy"
                dateFormatCalendar="LLLL yyyy"
                disabled={false}
                disabledKeyboardNavigation={false}
                dropdownMode="scroll"
                inlineFocusSelectedMonth={false}
                monthsShown={1}
                nextMonthButtonLabel="Next month"
                onBlur={[Function]}
                onChange={[Function]}
                onClickOutside={[Function]}
                onFocus={[Function]}
                onInputClick={[Function]}
                onInputError={[Function]}
                onKeyDown={[Function]}
                onMonthChange={[Function]}
                onSelect={[Function]}
                onYearChange={[Function]}
                preventOpenOnFocus={false}
                previousMonthButtonLabel="Previous Month"
                readOnly={false}
                renderDayContents={[Function]}
                shouldCloseOnSelect={true}
                showMonthYearPicker={false}
                showTimeInput={false}
                showTimeSelect={false}
                strictParsing={false}
                timeCaption="Time"
                timeInputLabel="Time"
                timeIntervals={30}
                withPortal={false}
              >
                <PopperComponent
                  hidePopper={true}
                  popperComponent={null}
                  popperModifiers={
                    Object {
                      "preventOverflow": Object {
                        "boundariesElement": "viewport",
                        "enabled": true,
                        "escapeWithReference": true,
                      },
                    }
                  }
                  popperPlacement="bottom-start"
                  popperProps={Object {}}
                  targetComponent={
                    <div
                      className="react-datepicker__input-container"
                    >
                      <input
                        className="transaction-line-date"
                        disabled={false}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        readOnly={false}
                        type="text"
                        value=""
                      />
                    </div>
                  }
                >
                  <Manager>
                    <Reference>
                      <InnerReference
                        setReferenceNode={[Function]}
                      >
                        <div
                          className="react-datepicker-wrapper"
                        >
                          <div
                            className="react-datepicker__input-container"
                          >
                            <input
                              className="transaction-line-date"
                              disabled={false}
                              onBlur={[Function]}
                              onChange={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              readOnly={false}
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </InnerReference>
                    </Reference>
                  </Manager>
                </PopperComponent>
              </DatePicker>
              <CurrencyInput
                allowNegative={false}
                autoFocus={false}
                className="transaction-line-amount"
                decimalSeparator="."
                inputType="text"
                onChange={[Function]}
                onChangeEvent={[Function]}
                precision="2"
                prefix=""
                selectAllOnFocus={false}
                suffix=""
                thousandSeparator=","
                value={1}
              >
                <input
                  className="transaction-line-amount"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onMouseUp={[Function]}
                  type="text"
                  value="1.00"
                />
              </CurrencyInput>
            </div>
          </div>
        </TransactionLine>
        <div
          className="col-sm-1"
        >
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            Submit
          </button>
        </div>
      </div>
    </AddTransactions>
    <TransactionsView
      transactions={
        Array [
          Object {
            "amount": 100,
            "category": "",
            "description": "",
          },
        ]
      }
    >
      <div>
        <TransactionBadge
          key="0"
          transaction={
            Object {
              "amount": 100,
              "category": "",
              "description": "",
            }
          }
        >
          <div
            className="list-group-item transaction-badge-amount"
          >
            1.00$
          </div>
        </TransactionBadge>
      </div>
    </TransactionsView>
  </div>
</App>
`;
